logic/src/commonMain/kotlin/com/wire/kalium/logic/data/team/TeamRepository.kt

```diff
@@ -172,4 +173,9 @@ internal class TeamDataSource(
             serviceDAO.insertMultiple(it)
         }
     }
+
+    override suspend fun approveLegalHold(teamId: TeamId, password: String?): Either<CoreFailure, Unit> = wrapApiRequest {
+        teamsApi.approveLegalHold(teamId.value, selfUserId.value, password)
+        // TODO: should we update the legal hold status for the current user in the database?
```

Comment:
I might not be in pair with the feature but I have some questions:
1. how/where do we use the legal hold status throughout the app currently?
2. not really a question, but, if we update on the database then we must keep track of it as well when it is changed, right? 🤔

Reply:
I think we should update clients of self user here, as legal state will be gotten from there by observing if tehre is any legal hold device